src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1660 @@
+ * of distinct possible states of such an object is called its <i>period</i>.
+ * (Some implementations of the {@link RandomGenerator} interface may be truly
+ * random rather than pseudorandom, for example relying on the statistical
+ * behavior of a physical object to derive chosen values. Such implementations
+ * do not have a fixed period.)
+ *
+ * <p> As a rule, objects that implement the {@link RandomGenerator} interface
+ * need not be thread-safe. It is recommended that multithreaded applications
+ * use either {@link ThreadLocalRandom} or (preferably) pseudorandom number
+ * generators that implement the {@link SplittableGenerator} or
+ * {@link JumpableGenerator} interface.
+ *
+ * <p> Objects that implement {@link RandomGenerator} are typically not
+ * cryptographically secure. Consider instead using {@link SecureRandom} to get
+ * a cryptographically secure pseudorandom number generator for use by
+ * security-sensitive applications. Note, however, that {@link SecureRandom}
+ * does implement the {@link RandomGenerator} interface, so that instances of
+ * {@link SecureRandom} may be used interchangeably with other types of
+ * pseudorandom generators in applications that do not require a secure
+ * generator.
+ *
+ * <p>Unless explicit stated otherwise, the use of null for any method argument
+ * will cause a NullPointerException.
+ *
+ * @since   17
+ *
+ */
+public interface RandomGenerator {
+     /**
+     * Returns an instance of {@link RandomGenerator} that utilizes the
+     * {@code name} algorithm.
+     *
+     * @param name  Name of random number generator algorithm
+     *
+     * @return An instance of {@link RandomGenerator}
+     *
+     * @throws NullPointerException if name is null
+     * @throws IllegalArgumentException if the named algorithm is not found
+     */
+    static RandomGenerator of(String name) {
+        Objects.requireNonNull(name);
+
+        return RandomGeneratorFactory.of(name, RandomGenerator.class);
+    }
+
+    /**
+     * Returns a {@link RandomGeneratorFactory} that can produce instances of
+     * {@link RandomGenerator} that utilize the {@code name} algorithm.
+     *
+     * @param name  Name of random number generator algorithm
+     *
+     * @return {@link RandomGeneratorFactory} of {@link RandomGenerator}
+     *
+     * @throws NullPointerException if name is null
+     * @throws IllegalArgumentException if the named algorithm is not found
+     */
+    static RandomGeneratorFactory<RandomGenerator> factoryOf(String name) {
+        Objects.requireNonNull(name);
+
+        return RandomGeneratorFactory.factoryOf(name, RandomGenerator.class);
+    }
+
+    /**
+     * Returns a {@link RandomGeneratorFactory} meeting the minimal requirement
+     * of having an algorithm whose state bits are greater than or equal 64.
+     *
+     * @implNote  Since algorithms will improve over time, there is no
+     * guarantee that this method will return the same algorithm each time.
```

Comment:
is there a guarantee that within a running jvm each invocation will return the same algorithm?

Reply:
In practice terms yes. The point is that the user should not assume they will be getting the same algorithm each time.